Runtime/BlockadeLabsSkyboxGenerator.cs

```diff
@@ -591,95 +591,184 @@ internal async Task DownloadResultAsync(GetImagineResult result, bool setSkybox
                 return;
             }
 
+            if (setSkybox && _skyboxMesh != null)
+            {
+                _skyboxMesh.SetMetadata(result);
+            }
+
             const int maxLength = 20;
 
             if (prompt.Length > maxLength)
             {
                 prompt = prompt.Substring(0, maxLength);
             }
 
+            SkyboxAI skyboxAI = null;
+            AssetUtils.CreateGenerateBlockadeLabsFolder();
             var prefix = AssetUtils.CreateValidFilename(prompt);
-            var folderPath = AssetUtils.CreateUniqueFolder(prefix);
-            var texturePath = $"{folderPath}/{prefix} texture.png";
-            var depthTexturePath = $"{folderPath}/{prefix} depth texture.png";
-            var resultsPath = $"{folderPath}/{prefix} data.txt";
 
-            var tasks = new List<Task>
+            if (!AssetUtils.TryCreateFolder(prefix, out var folderPath))
             {
-                ApiRequests.DownloadFileAsync(textureUrl, texturePath)
-            };
+                // check if there is a .txt file. If it exists then convert it to a skyboxAI asset and delete it
+                // check all directories that start with the prefix, but contain the *data.txt
+                var dataFiles = Directory.GetFiles(Application.dataPath, $"*{prefix} data.txt", SearchOption.AllDirectories).ToList();
+                // check if there are any existing skyboxAI assets with the same id
+                dataFiles.AddRange(Directory.GetFiles(Application.dataPath, $"{prefix}.asset", SearchOption.AllDirectories));
 
-            if (hasDepthMap)
-            {
-                tasks.Add(ApiRequests.DownloadFileAsync(depthMapUrl, depthTexturePath));
-            }
+                foreach (var dataFile in dataFiles)
+                {
+                    var currentDirectoryPath = Directory.GetParent(Path.GetFullPath(dataFile))!.FullName;
+                    var currentDirectorySkyboxAIPath = $"{currentDirectoryPath}/{prefix}.asset";
+
```

Comment:
I think the foreach is wrong here. I think there can be multiple data files with the same name, but in different folders (because we were using CreateUniqueFolder, but the assets inside might have non-unique names)

Reply:
I think if we can't find the right folder by name or the data ID doesn't match, then just don't do any upgrade and create a new folder.

Reply:
I tested this out a few different ways. Pretty sure it is working as intended. If you can repro a legitimate way it breaks let me know. Otherwise I'd prefer we always upgrade to SkyboxAI.asset